src/Cafe/OS/libs/nsyshid/BackendLibusb.h

```diff
@@ -44,6 +44,11 @@ namespace nsyshid::backend::libusb
 										bool& endpointOutFound, uint8& endpointOut, uint16& endpointOutMaxPacketSize);
 	};
 
+	template<typename T>
+	using UniquePtr = std::unique_ptr<T, void (*)(T*)>;
+
+	using ConfigDescriptor = UniquePtr<libusb_config_descriptor>;
```

Comment:
That's such a neat way to deal with C memory management :)

Reply:
Wish I could claim this code as my own but this os borrowed from another libusb implementation